### PR TITLE
add kitty-review command

### PR DESCRIPTION
**Pull request by petulkauwu**
Treba jsem tupej, a toto nefakci. 

Kazdopadne, udelej si API key na gify.

### CODE REVIEW

File: src/commands/kitty-review.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        .send()
+        .await;
+
+    if let Ok(response) = response {
+        if let Ok(json) = response.json::<serde_json::Value>().await {
+            if let Some(results) = json.get("results").and_then(|results| results.as_array()) {
+                if let Some(random_result) = results.choose(&mut rand::thread_rng()) {
+                    if let Some(media) = random_result.get("media").and_then(|media| media.as_array()) {
+                        if let Some(first_media) = media.get(0) {
+                            if let Some(gif) = first_media.get("gif").and_then(|gif| gif.get("url").and_then(|url| url.as_str())) {
```
Comment (marekvospel) [inline src/commands/kitty-review.rs:44]: Unnecessary to do so many json operations and Option checks, It's smarter to create a struct (with Serialize and Deserialize derive) and check whether the custom result output was parsed successfully.

https://serde.rs/derive.html
See also /src/state/config.rs

File: src/commands/kitty-review.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    match locale.as_str() {
+        "cs" => t!("commands.kitty-review.message", "cs"),
+        _ => t!("commands.kitty-review.message"),
+    }
+}
+
+pub(crate) async fn run(ctx: &Context, command: &ApplicationCommandInteraction) {
+    let locale = command.clone().locale;
+
+    let tenor_api_key = "API KEY ZDE";
```
Comment (marekvospel) [inline src/commands/kitty-review.rs:30]: API KEY should NEVER be part of your source code, you can either use environment or use the config object (see how discord_token is loaded from Config.toml)

File: src/commands/kitty-review.rs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    let search_url = "https://tenor.googleapis.com/v2/search?q=kitty-review&key=".to_owned() + tenor_api_key;
+    let response = reqwest::Client::new()
+        .get(&search_url)
+        .header(header::USER_AGENT, "SSPS-KB BOT")
```
Comment (marekvospel) [inline src/commands/kitty-review.rs:34]: `User-Agent: <product> / <product-version> <comment>`
```suggestion
        .header(header::USER_AGENT, "SSPS-KB/1.0 workshop-bot")
```

File: src/commands/kitty-review.rs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        .send()
+        .await;
```
Comment (marekvospel) [inline src/commands/kitty-review.rs:36]: This code should be in a separate function, so it can be reused.

File: src/commands/kitty-review.rs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                                                    .embed(|embed| embed
+                                                        .image(gif)
+                                                        .color(Color::from_rgb(110, 110, 110)))
+                                            })
+                                    })
+                                    .await
+                                {
+                                    error!("There was an error while responding to kitty-review command: {}", e)
+                                }
+                                return;
```
Comment (marekvospel) [inline src/commands/kitty-review.rs:61]: unnecessary return

File: src/commands/kitty-review.rs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                            }
+                        }
+                    }
+                }
+            }
+        }
```
Comment (marekvospel) [inline src/commands/kitty-review.rs:67]: No error log

**Review by marekvospel — CHANGES_REQUESTED** [review]

**Comment by marekvospel** [comment]
Also please create a change file inside .changes directory

**Comment by petulkauwu** [comment]
moje rust skills above

**Comment by petulkauwu** [comment]
taky brainless code

**Comment by marekvospel** [comment]
@petulkauwu I need you to actually do the requested changes, not just click resolve

**Comment by petulkauwu** [comment]
ono se to a me to napsalo ze se to samo commitne :(

**Comment by petulkauwu** [comment]
udělám pak novy

**Comment by petulkauwu** [comment]
very real!!!!!

**Pull request closed without merging**